llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -1366,7 +1366,8 @@ bool GCNPassConfig::addPreISel() {
   // control flow modifications.
   addPass(createAMDGPURewriteUndefForPHILegacyPass());
 
-  addPass(createLCSSAPass());
+  if (!getCGPassBuilderOption().EnableGlobalISelOption)
```

Comment:
You can't just disable a pass for only globalisel. This will break the dag if there is a fallback